src/main/java/com/fvv/bookstore/dao/OrderItemDAOImpl.java

```diff
@@ -55,4 +62,197 @@ public void addOrderItemInBatch(final List<OrderItem> orderItems) throws DaoExce
 			}
 		}		
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<OrderItem> findBookItemsByOrder(final Long idOrder) throws DaoException, OrderNotFoundException {
+		List<OrderItem> orderItems = new ArrayList<>();
+		try (
+				Connection conn = ConnectionFactory.getConnection(); 
+				PreparedStatement ps = conn.prepareStatement(SqlQueryEnum.ORDER_ITEM_BOOK_SELECT_ORDER_ID.getQuery())
+		) {
+			ps.setLong(1, idOrder);
+			try (ResultSet rs = ps.executeQuery()) {
+				while (rs.next()) {
+					Book book = new Book();
+					book.setId(rs.getLong("ord_ite_book_book_id"));
+					book.setTitle(rs.getString("book_title"));
+					book.setUnitPrice(rs.getDouble("book_unit_price"));
+					
+					OrderItem bookItem = new OrderItem();
+					bookItem.setProduct(book);
+					
+					Order order = new Order();
+					order.setId(idOrder);
+					bookItem.setOrder(order);
+					
+					bookItem.setQuantity(rs.getInt("ord_ite_book_quantity"));
+					bookItem.setItemAmount(rs.getDouble("ord_ite_book_amount"));
+					
+					orderItems.add(bookItem);
+				} 
+			}
+		} catch(SQLException e) {
+			throw new DaoException("Error to find an order", e);
+		} 
+		return orderItems;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<OrderItem> findCellphoneItemsByOrder(final Long idOrder) throws DaoException, OrderNotFoundException {
+		List<OrderItem> orderItems = new ArrayList<>();
+		try (
+				Connection conn = ConnectionFactory.getConnection(); 
+				PreparedStatement ps = conn.prepareStatement(SqlQueryEnum.ORDER_ITEM_CELLPHONE_SELECT_ORDER_ID.getQuery())
+		) {
+			ps.setLong(1, idOrder);
+			try (ResultSet rs = ps.executeQuery()) {
+				while (rs.next()) {
+					Cellphone cellphone = new Cellphone();
+					cellphone.setId(rs.getLong("ord_ite_cel_cel_id"));
+					cellphone.setBrand(rs.getString("cel_brand"));
+					cellphone.setUnitPrice(rs.getDouble("cel_unit_price"));
+					
+					OrderItem celItem = new OrderItem();
+					celItem.setProduct(cellphone);
+					
+					Order order = new Order();
+					order.setId(idOrder);
+					celItem.setOrder(order);
+					
+					celItem.setQuantity(rs.getInt("ord_ite_cel_quantity"));
+					celItem.setItemAmount(rs.getDouble("ord_ite_cel_amount"));
+					
+					orderItems.add(celItem);
+				} 
+			}
+		} catch(SQLException e) {
+			throw new DaoException("Error to find an order", e);
+		} 
+		return orderItems;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<OrderItem> findDvdtemsByOrder(final Long idOrder) throws DaoException, OrderNotFoundException {
+		List<OrderItem> orderItems = new ArrayList<>();
+		try (
+				Connection conn = ConnectionFactory.getConnection(); 
+				PreparedStatement ps = conn.prepareStatement(SqlQueryEnum.ORDER_ITEM_DVD_SELECT_ORDER_ID.getQuery())
+		) {
+			ps.setLong(1, idOrder);
+			try (ResultSet rs = ps.executeQuery()) {
+				while (rs.next()) {
+					Dvd dvd;
+					if(rs.getString("dvd_movie_director") != null) {
+						MovieDvd dvdMovie = new MovieDvd();
```

Comment:
this variable is redundant, you can instantiate directly in the line below

Reply:
ok